_posts/en/newsletters/2019-05-28-newsletter.md

```diff
@@ -0,0 +1,593 @@
+
+{% comment %}<!-- include references.md below the fold but above any Jekyll/Liquid variables-->{% endcomment %}
+{% include references.md %}
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **Proposed new opcode for transaction output commitments:** Jeremy
+  Rubin posted to the Bitcoin-Dev mailing list a proposal to soft fork
+  in an `OP_CHECKOUTPUTSHASHVERIFY` opcode that allows a Bitcoin address
+  to require the transaction spending it include a certain set of
+  outputs.  For details, please see the special section about the
+  proposal later in this newsletter.
+
+- **Continued discussion about bip-taproot and bip-tapscript:**
+  two comments this week from the Bitcoin-Dev mailing list discussion
+  seemed particularly noteworthy:
+
+    - *Final stack empty:* in legacy, segwit, and proposed
+      [bip-tapscript][] scripts, a script evaluates successfully if it
+      contains exactly[^legacy-cleanstack] one element that is *true*.
+      Russell O'Connor [raised][emptystack1] a point he's [raised
+      before][emptystack0] and requested that this opportunity be taken
+      to require tapscript only evaluate successfully if it ends with an
+      empty stack.  Pieter Wuille [replied][emptystack reply] that his
+      work on [miniscript][] (see [Newsletter #32][]) showed that, for
+      the subset of scripts miniscript will create, this change in
+      semantics will at most save 0.25 vbytes per tapscript.  Also,
+      although the change may also simplify development for anyone
+      writing scripts by hand, it's a bit risky as every development
+      guide to Script written to date teaches that scripts must
+      terminate with a *true* value on the stack.  Wuille summarized,
+      "so overall this feels like something with marginal costs, but
+      also at most marginal benefits."
+
+    - *Move the oddness byte:* Bitcoin public keys are graph points, so
+      they are most naturally specified using an X,Y coordinate pair.
+      However, because a valid pubkey must be on an [elliptic curve][],
+      it's possible to find the Y coordinate for any given X coordinate
+      given the curve formula and a single bit of additional data.
+      Because there's no convenient place in Bitcoin transactions to add
+      just a single additional bit of data per output, Taproot addresses
+      use an entire additional byte for this bit.
+
+        This week, John Newbery [suggested][smaller v1 spk] that we use some method to
+        avoid placing this byte in the scriptPubKey.  Wuille agreed that
+        this could be useful and will attempt implementing a variation
+        where the bit will be included as part of the taproot witness
+        data.  This will reduce the cost create a taproot output by 1.0
+        vbytes (making it the same as P2WSH currently) with either zero
+        or 0.25 vbytes additional cost to spend a taproot output.
+
+- FIXME maybe minisketch thing
```

Comment:
Better to give things at least a day or two before a write up in general, in my opinion; better if the newsletter tries to be detailed and informative rather than catching breaking news. YMMV etc